### DISCUSSION TITLE
ideaMaker  Dreamer NX Start End G Codes examples

### DISCUSSION THREAD

**Discussion by Lucien-X** [moonglow/FlashForge_Marlin#45] [Ideas]
--------------------------------------------------------------
[ideaMaker](https://www.raise3d.com/ideamaker/)

[Introduction by DrVax](https://www.youtube.com/watch?v=xL_IDura2C0)

[FF DNX ideamaker configs.zip](https://github.com/moonglow/FlashForge_Marlin/files/6900646/FF.DNX.ideamaker.configs.zip)

Start:
```gcode
;Start Gcode
G90
M140 S{temperature_heatbed}     ;Heat bed up to first layer temperature
M190 S{temperature_heatbed}     ;Wait for Heat Bed temperature
M104 S{temperature_extruder1}   ;Set nozzle temperature to first layer temperature
M109 S{temperature_extruder1}    ;Wait for Extruder temperature
M107 ;start with the fan off
G28
M117 Purge extruder...
G1 Z{print_pos_min_z} F420
;Purge line
G1 X-110.00 Y-60.00 F4800
G1 Z{print_pos_min_z} F420
G92 E0
G1 X-110.00 Y60.00 E17,4 F1200
;Purge line end
G92 E0
G1 Z2.0 F3000 ; Move Z Axis up little to prevent scratching of Heat Bed
M117 Printing...
```
End:
```gcode
;end gcode
M140 S0 ; turn off heatbed
M104 S0 ; turn off temperature
G28 X Y
M84 X Y E ; disable motors
```
--------------------------------------------------------------

**Comment by Lucien-X** [comment]
@moonglow This snippet maybe helpful with people who always wanna try some thing new.
Could you append this in your documention below?
https://github.com/moonglow/FlashForge_Marlin/wiki/FlashForge-Dreamer-NX-Start-End-G-Codes-examples

**Reply by moonglow** [reply]
Done ! Thank you
